requirements-dev.in

```diff
@@ -2,6 +2,8 @@
 
 debugpy
 pydot==1.4.2  # ERD diagram
+pandas-stubs
+sqlalchemy-stubs
```

Comment:
Could we move these down below `mypy` so that they're all grouped logically together? And can we also add all of the packages that mypy wants (in a clean venv, run `mypy .`, then `mypy --install-types` to get the list.

These dependencies should be in sync between this file and the `.pre-commit-config.yml` mypy config.

Reply:
Added dependencies to pre-commit-config.yaml, and also to requirements-dev.in